src/vs/workbench/contrib/issue/electron-sandbox/nativeIssueFormService.ts

```diff
@@ -0,0 +1,71 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import 'vs/css!./media/newIssueReporter';
+import { IMenuService } from 'vs/platform/actions/common/actions';
+import { IContextKeyService } from 'vs/platform/contextkey/common/contextkey';
+import { IDialogService } from 'vs/platform/dialogs/common/dialogs';
+import { INativeEnvironmentService } from 'vs/platform/environment/common/environment';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { ILogService } from 'vs/platform/log/common/log';
+import { INativeHostService } from 'vs/platform/native/common/native';
+import product from 'vs/platform/product/common/product';
+import { BrowserWindow } from 'vs/workbench/browser/window';
+import { IssueFormService } from 'vs/workbench/contrib/issue/browser/issueFormService';
+import { IIssueFormService, IssueReporterData } from 'vs/workbench/contrib/issue/common/issue';
+import { IssueReporter2 } from 'vs/workbench/contrib/issue/electron-sandbox/issueReporterService2';
+import { IAuxiliaryWindowService } from 'vs/workbench/services/auxiliaryWindow/browser/auxiliaryWindowService';
+import { IHostService } from 'vs/workbench/services/host/browser/host';
+
+export class NativeIssueFormService extends IssueFormService implements IIssueFormService {
+	private issueReporterParentWindow: BrowserWindow | null = null;
+
+	constructor(
+		@IInstantiationService instantiationService: IInstantiationService,
+		@IAuxiliaryWindowService auxiliaryWindowService: IAuxiliaryWindowService,
+		@ILogService logService: ILogService,
+		@IDialogService dialogService: IDialogService,
+		@IMenuService menuService: IMenuService,
+		@IContextKeyService contextKeyService: IContextKeyService,
+		@IHostService hostService: IHostService,
+		@INativeHostService private readonly nativeHostService: INativeHostService,
+		@INativeEnvironmentService private readonly environmentService: INativeEnvironmentService,) {
```

Comment:
is IHostService not the same as INativeHostService in this context? Can you just grab the INativeHostService and that pass that in to the `super` as the `IHostService` parameter?

Reply:
hmmm, we need `INativeHostService` since it gets us OS stuff, but I can't pass it into the base `IssueFormService` since it's `browser`

Reply:
Fair, I guess you really only need a service that does this:
```ts
reload(options?: { disableExtensions?: boolean }): Promise<void>;
```

both the `IHostService` and `INativeHostService` do this so your base class could ask for an object that looks like:
```ts
{ reload(options?: { disableExtensions?: boolean }): Promise<void>; }
```

and then in this ctor, pass in `nativeHostService` instead of `IHostService`.

Reply:
It's odd to me that `INativeHostService` doesn't extend `IHostService`... in any case, this is a suggestion

Reply:
gotcha, I tried it out and it gives me a bunch of errors in the contribution since i tried passing in the reload object instead:

```
Argument of type 'typeof IssueFormService' is not assignable to parameter of type 'new (...services: BrandedService[]) => IIssueFormService'.
  Types of construct signatures are incompatible.
    Type 'new (instantiationService: IInstantiationService, auxiliaryWindowService: IAuxiliaryWindowService, menuService: IMenuService, contextKeyService: IContextKeyService, logService: ILogService, dialogService: IDialogService, hostService: { ...; }) => IssueFormService' is not assignable to type 'new (...services: BrandedService[]) => IIssueFormService'.
      Types of parameters 'instantiationService' and 'services' are incompatible.
        Type 'BrandedService' is missing the following properties from type 'IInstantiationService': createInstance, invokeFunction, createChild, disposets(2345)
```

maybe will leave as is for now? -  nativeHostService now mainly for getting OS properties, hostService used in both for reload in the base class